From b554198d3f882ad7d6637ba219593bff6afd6d68 Mon Sep 17 00:00:00 2001
From: Vincent Penquerc'h <vincent.penquerch@collabora.co.uk>
Date: Wed, 21 Sep 2011 14:53:01 +0100
Subject: [PATCH 132/154] mpegpsdemux: take into account SCR offset when
 seeking

Since the seeking byte offset is chosen by linear interpolation
from SCR values, we need to take that first SCR into account
to end up near the correct offset. Otherwise, as the code does
a linear search after that first seek, it will take a LOOOOOONG
time to get there for streams which don't start at zero.

https://bugzilla.gnome.org/show_bug.cgi?id=659485
---
 gst/mpegdemux/gstmpegdemux.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/gst/mpegdemux/gstmpegdemux.c b/gst/mpegdemux/gstmpegdemux.c
index ca2bac4..d1eef4b 100644
--- a/gst/mpegdemux/gstmpegdemux.c
+++ b/gst/mpegdemux/gstmpegdemux.c
@@ -1034,8 +1034,9 @@ gst_flups_demux_do_seek (GstFluPSDemux * demux, GstSegment * seeksegment)
   GST_INFO_OBJECT (demux, "sink segment configured %" GST_SEGMENT_FORMAT
       ", trying to go at SCR: %" G_GUINT64_FORMAT, &demux->sink_segment, scr);
 
-  offset = MIN (gst_util_uint64_scale (scr, scr_rate_n, scr_rate_d),
-      demux->sink_segment.stop);
+  offset =
+      MIN (gst_util_uint64_scale (scr - demux->first_scr, scr_rate_n,
+          scr_rate_d), demux->sink_segment.stop);
 
   found = gst_flups_demux_scan_forward_ts (demux, &offset, SCAN_SCR, &fscr);
   if (!found) {
-- 
1.7.5.4

